modules/fonts/default.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+with builtins;
+
+let
+  cfg = config.fonts;
+  readDirsRec = path: let
+    home = readDir path;
+    list = mapAttrsToList (name: type:
+      let newPath = "${path}/${name}";
+      in if (type == "directory" || type == "symlink") && !(isFont name) then readDirsRec newPath else [ newPath ]) home;
+    in flatten list;
+  isFont = name: let
+    fontExt = [ ".ttf" ".otf" ];
+    hasExt = exts: name: foldl' (acc: ext: (hasSuffix ext name) || acc) false exts;
+    in hasExt fontExt name;
+  fontFiles = dir: filter isFont (readDirsRec dir);
+  libraryLink = font: "ln -fn '${font}' '/Library/Fonts/${baseNameOf font}'";
+  outLink = font: "ln -sfn -t $out/share/fonts/ '${font}'";
+  fontLinks = link: dir: concatMapStringsSep "\n" link (fontFiles dir);
+  systemFontsDir = pkgs.runCommand "systemFontsDir" {} ''
```

Comment:
The benefit of this change is that the enableFontsDir will work exactly as in NixOS. We could now try to create an intermediate step with linking through `/etc/static` or directly through `/run/current-system/sw/share/fonts`. @LnL7 thoughts?